src/main/rx/cc2500_sfhss.c

```diff
@@ -0,0 +1,448 @@
+    DEBUG_SET(DEBUG_RX_SFHSS_SPI, DEBUG_DATA_OFFSET_MIN, bindOffset_min);
+    cc2500WriteReg(CC2500_0C_FSCTRL0, (uint8_t)bindOffset_min);
+    sfhss_channel = BIND_CH;
+    sfhssRx();
+}
+
+static bool tune1Rx(uint8_t *packet)
+{
+    if (bindOffset_min >= 126) {
+        bindOffset_min = -126;
+        DEBUG_SET(DEBUG_RX_SFHSS_SPI, DEBUG_DATA_OFFSET_MIN, bindOffset_min);
+    }
+    if ((millis() - timeTunedMs) > 220) {   // 220ms
+        timeTunedMs = millis();
+        bindOffset_min += BIND_TUNE_STEP<<2;
+        DEBUG_SET(DEBUG_RX_SFHSS_SPI, DEBUG_DATA_OFFSET_MIN, bindOffset_min);
+        cc2500WriteReg(CC2500_0C_FSCTRL0, (uint8_t)bindOffset_min);
+        cc2500Strobe(CC2500_SRX);
+    }
+    if (sfhssRecv(packet)){
+        if (sfhssPacketParse(packet, false)){
+            if ((packet[SFHSS_PACKET_LEN - 1] & 0x7F) > 40 ){    /* lqi */
+                rxSfhssSpiConfigMutable()->bindTxId[0] = GET_TXID1(packet);
+                rxSfhssSpiConfigMutable()->bindTxId[1] = GET_TXID2(packet);
+                bindOffset_max = bindOffset_min;
+                DEBUG_SET(DEBUG_RX_SFHSS_SPI, DEBUG_DATA_OFFSET_MAX, bindOffset_max);
+                cc2500Strobe(CC2500_SRX);
+                timeTunedMs = millis();
+                return true;
+            }
+        }
+        cc2500Strobe(CC2500_SRX);
+    }
+    return false;
+}
+
+static bool tune2Rx(uint8_t *packet)
+{
+    cc2500LedBlink(100);
+    if (((millis() - timeTunedMs) > 880) || bindOffset_max > (126-BIND_TUNE_STEP)) {   // 220ms *4
+        timeTunedMs = millis();
+        cc2500WriteReg(CC2500_0C_FSCTRL0, (uint8_t)bindOffset_min);
+        cc2500Strobe(CC2500_SRX);
+        return true;
+    }
+    if (sfhssRecv(packet)){
+        if (sfhssPacketParse(packet, true)){
+            timeTunedMs = millis();
+            bindOffset_max += BIND_TUNE_STEP;
+            DEBUG_SET(DEBUG_RX_SFHSS_SPI, DEBUG_DATA_OFFSET_MAX, bindOffset_max);
+            cc2500WriteReg(CC2500_0C_FSCTRL0, (uint8_t)bindOffset_max);
+        }
+        cc2500Strobe(CC2500_SRX);
+    }
+    return false;
+}
+
+static bool tune3Rx(uint8_t *packet)
+{
+    cc2500LedBlink(100);
+    if (((millis() - timeTunedMs) > 880) || bindOffset_min < (-126+BIND_TUNE_STEP)) {   // 220ms *4
+        return true;
+    }
+    if (sfhssRecv(packet)){
+        if (sfhssPacketParse(packet, true)){
+            timeTunedMs = millis();
+            bindOffset_min -= BIND_TUNE_STEP;
+            DEBUG_SET(DEBUG_RX_SFHSS_SPI, DEBUG_DATA_OFFSET_MIN, bindOffset_min);
+            cc2500WriteReg(CC2500_0C_FSCTRL0, (uint8_t)bindOffset_min);
+        }
+        cc2500Strobe(CC2500_SRX);
+    }
+    return false;
+}
+
+void sfhssnextChannel(void)
+{
+    do {
+        sfhss_channel += sfhss_code + 2;
+        if (sfhss_channel > 29) {
+            sfhss_channel -= 31;
+        }
+    }while( sfhss_channel < 0);
+
+    sfhssRx();
+}
+
+void sfhssSpiSetRcData(uint16_t *rcData, const uint8_t *payload)
+{
+    if( GET_COMMAND(payload) & 0x8 ){ 
```

Comment:
There is a trailing space after '& 0x8 ){'.

Reply:
fixed